### PR TITLE
add `lazy_normalization_consts` feature gate

### PR DESCRIPTION
**Pull request by lcnr**
In #71973 I underestimated the amount of code which is influenced by lazy normalization of consts
and decided against having a separate feature flag for this.

Looking a bit more into this, the following issues are already working with lazy norm in its current state #47814 #57739 #73980

I therefore think it is worth it to enable lazy norm separately. Note that `#![feature(const_generics)]` still automatically activates
this feature, so using `#![feature(const_generics, lazy_normalization_consts)]` is redundant.

r? @varkor @nikomatsakis

### CODE REVIEW

File: src/librustc_feature/active.rs [hunk 1]
```diff
@@ -567,6 +567,9 @@ declare_features! (
     /// Be more precise when looking for live drops in a const context.
     (active, const_precise_live_drops, "1.46.0", Some(73255), None),
 
+    /// Lazily evaluate constants. Which allows constants to depend on type parameters.
```
Comment (varkor) [inline src/librustc_feature/active.rs:570]:
```suggestion
    /// Lazily evaluate constants. This allows constants to depend on type parameters.
```

**Comment by varkor** [comment]
r=me with comment fixed.

**Comment by lcnr** [comment]
@bors r=varkor

I forgot to incorporate the comment fix in my rebase :sweat_smile:

**Pull request merged**